guides/messaging-clustering.adoc

```diff
@@ -0,0 +1,174 @@
+= Configuring Clustered Messaging in WildFly
+:summary: Learn how to configure WildFly with an ActiveMQ Artemis Cluster, Server-Side Message and Client Load Balancing.
+:includedir: _includes
+include::{includedir}/_attributes.adoc[]
+// you can override any attributes eg to lengthen the
+// time to complete the guide
+:prerequisites-time: 15
+
+In this guide, you will learn how to configure WildFly servers with an embedded ActiveMQ Artemis broker in a messaging cluster,
+enabling server-side message and client load balancing. It will demonstrate how to configure deployments, such as MDBs,
+EJBs, or Servlets, as well as external JMS clients, to load balance connections across all servers in the cluster.
+
+[NOTE]
+--
+An ActiveMQ Artemis cluster is distinct from a standard WildFly cluster formed by the Infinispan subsystem.
+While a WildFly cluster using Infinispan focuses on data caching and session replication across nodes, an Artemis cluster
+is specifically designed for messaging, allowing message redistribution across brokers and load balancing of message producers and consumers.
+These two clusters operate independently, each with its own purpose and configuration, although they can coexist within
+the same WildFly environment.
+--
+
+For a detailed explanation of ActiveMQ Artemis clusters, refer to the https://activemq.apache.org/components/artemis/documentation/latest/clusters.html#overview[Apache ActiveMQ Artemis Clustering Documentation]
+and the https://docs.redhat.com/en/documentation/red_hat_jboss_enterprise_application_platform/7.4/html-single/configuring_messaging/index#clusters_overview[EAP 7.4 Messaging Clusters Overview].
+
+[[prerequisites]]
+== Prerequisites
+
+To complete this guide, you need:
+
+* Roughly {prerequisites-time} minutes
+* JDK {jdk-minimal-version} installed with `JAVA_HOME` configured appropriately
+
+== Prepare WildFly Servers
+
+Now, let's configure two WildFly servers with embedded ActiveMQ Artemis brokers in a cluster. We'll use the `full-ha` profile,
+as it already contains most of the necessary settings and requires only minor modifications.
+
+By default, the messaging cluster in WildFly uses JGroups, which relies on a multicast-based discovery mechanism to locate
+servers in the cluster. This allows the cluster to automatically discover and join new servers within the network. However,
```

Comment:
This allows the cluster to automatically discover and join new servers within the network.I would have written **add** instead of **join**

Reply:
Thanks, fixed.